### PR TITLE
Step-9: Adding Toast Message

### PR DESCRIPTION
**Pull request by fza-wtag**
## Description
[Step-9](https://trello.com/c/dTRcid0J)

This PR introduces all changes regarding adding toast message notifications.

## Checklist

- [x]  Toast messages are shown whenever a new task is created, edited, deleted, and searched.
- [x]  Warning toast is also added for not having a title.

## Attachments
https://user-images.githubusercontent.com/119289799/218303388-dad66c0f-8cb1-4554-9eea-94d9ebd6a9ee.mov

### CODE REVIEW

File: todos/src/App.js [hunk 1]
```diff
@@ -4,12 +4,14 @@ import Todos from "components/Todos";
 import { Provider } from "react-redux";
 import store from "store";
 import TopContent from "components/TopContent";
+import { ToastContainer } from "react-toastify";
 
 function App() {
   return (
     <div>
```
Comment (hhu-wtag) [inline todos/src/App.js:11]: Can we use React Fragment here?

File: todos/src/components/AddTask.js [hunk 2]
```diff
@@ -8,6 +8,10 @@ import {
   toggleAddTaskVisibility,
   toggleAddTaskButtonVisibility,
 } from "actions/index";
+import { infoMessage, successMessage, warningMessage } from "toast_methods";
+import { TASK_ADDED_MESSAGE } from "constants";
+import { EMPTY_TITLE_MESSAGE } from "constants";
+import { CREATE_CANCEL_MESSAGE } from "constants";
```
Comment (hhu-wtag) [inline todos/src/components/AddTask.js:14]: We can import them in one line

File: todos/src/components/AddTask.js [hunk 3]
```diff
@@ -21,7 +25,7 @@ const AddTask = () => {
   const [inputData, setInputData] = useState("");
   const dispatch = useDispatch();
 
-  const handleClick = () => {
+  const handleSateChange = () => {
```
Comment (hhu-wtag) [inline todos/src/components/AddTask.js:28]: typo

File: todos/src/components/AddTask.js [hunk 4]
```diff
@@ -30,7 +34,8 @@ const AddTask = () => {
       //to get the job done with enter button
       dispatch(addTodo(inputData.slice(0, -1)), setInputData(""));
       if (inputData !== "") {
-        handleClick();
+        successMessage(TASK_ADDED_MESSAGE);
+        handleSateChange();
```
Comment (hhu-wtag) [inline todos/src/components/AddTask.js:38]: ditto

File: todos/src/components/CompletedTask.js [hunk 5]
```diff
@@ -2,6 +2,8 @@ import React from "react";
 import del from "icons/delete.svg";
 import { useDispatch } from "react-redux";
 import { deleteTodo } from "actions/index";
+import { dangerMessage } from "toast_methods";
```
Comment (hhu-wtag) [inline todos/src/components/CompletedTask.js:5]: We can make the helper method name, camelCased

File: todos/src/components/EditTask.js [hunk 6]
```diff
@@ -4,6 +4,10 @@ import done from "icons/done.svg";
 import "styles/addTask.css";
 import { useDispatch } from "react-redux";
 import { editUpdateCompleted, updateTodo } from "actions/index";
+import { successMessage, infoMessage } from "toast_methods";
+import { STATE_CHANGE_MESSAGE } from "constants";
+import { EDIT_CANCEL_MESSAGE } from "constants";
+import { EDIT_SUCCESS_MESSAGE } from "constants";
```
Comment (hhu-wtag) [inline todos/src/components/EditTask.js:10]: ditto

File: todos/src/components/IncompleteTask.js [hunk 7]
```diff
@@ -2,17 +2,22 @@ import React from "react";
 import del from "icons/delete.svg";
 import done from "icons/done.svg";
 import edit from "icons/edit.svg";
-import { useDispatch} from "react-redux";
-import { deleteTodo, updateCompleted, editTodo } from "actions/index";;
+import { useDispatch } from "react-redux";
+import { deleteTodo, updateCompleted, editTodo } from "actions/index";
+import { dangerMessage, successMessage } from "toast_methods";
+import { TASK_DELETE_MESSAGE } from "constants";
+import { STATE_CHANGE_MESSAGE } from "constants";
```
Comment (hhu-wtag) [inline todos/src/components/IncompleteTask.js:9]: ditto

File: todos/src/constants.js [hunk 8]
```diff
@@ -1,3 +1,11 @@
...
 export const SHOW_LESS = "Show Less";
 export const PER_PAGE = 4;
+export const TASK_ADDED_MESSAGE = "New task added successfully 🚀";
+export const EMPTY_TITLE_MESSAGE = "Please enter the task name/ title 😅";
+export const CREATE_CANCEL_MESSAGE = "Creating new task got cancelled 🙂";
+export const EDIT_CANCEL_MESSAGE = "Task editing got canceled 🙂";
+export const TASK_DELETE_MESSAGE = "Task deleted successfully 🙌";
+export const STATE_CHANGE_MESSAGE = "Task state changed to completed 💪";
+export const EDIT_SUCCESS_MESSAGE = "Task edited successfully  📝";
+export const SEARCH_DATA_MESSAGE= "Search data fetched 🕵️‍♀️"
```
Comment (hhu-wtag) [inline todos/src/constants.js:11]: Please refer to the design for more refined messages

File: todos/src/toast_methods.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
+import { toast, Zoom } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+
+const commonStyles = {
+  position: "bottom-right",
```
Comment (hhu-wtag) [inline todos/src/toast_methods.js:5]: position should be top-middle

**Review by hhu-wtag — CHANGES_REQUESTED** [review]

File: todos/src/constants.js (hunk 8 above)
Comment (fza-wtag) [reply todos/src/constants.js:11]: @hhu-wtag Bhai, does that means I need to make my toast messages as per the UI design?

File: todos/src/App.js (hunk 1 above)
Comment (fza-wtag) [reply todos/src/App.js:11]: Yes, @hhu-wtag Bhai. Like so?
```
<React.Fragment>
      <Provider store={store}>
        <Navbar />
        <ToastContainer />
        <TopContent />
        <Todos />
      </Provider>
    </React.Fragment>
```

Comment (hhu-wtag) [reply todos/src/App.js]: Yes

**Review by hhu-wtag — APPROVED** [review]

**Review by ray-wtag — APPROVED** [review]

**Pull request merged**